tests/helpers/test_service.py

```diff
@@ -381,6 +381,7 @@ async def test_extract_entity_ids_from_area(hass, area_mock):
     assert {
         "light.in_area",
         "light.assigned_to_area",
+        "light.in_other_area",
```

Comment:
This is actually incorrect.


See mock:
```py
entity_in_other_area = ent_reg.RegistryEntry(
    entity_id="light.in_other_area",
    unique_id="in-other-area-id",
    platform="test",
    device_id=device_in_area.id,
    area_id="other-area",
)
```

While this device might be in that area, the area for the entity is overridden (and thus should not show up).

Reply:
Ok, expected this. I'll revert that.